### PR TITLE
Add react-native-unimodules and move RN platform folders up a level

### PR DESCRIPTION
**Pull request by buoyad**
- [x] ios setup
- [x] android setup
- [x] move `react-native/android/` to `android/` to match classical RN setup
- [x] ? move `react-native/ios` to `ios/`
- add `expo-sms`(next pr)
- PoC use to open team invite message (next pr)
---
- [x] ios test build
- [x] android test build
- [x] ios storybook
- [x] android storybook

### CODE REVIEW

**Comment by buoyad** [comment]
@keybase/react-hackers this is ready!

File: shared/android/settings.gradle [hunk 1]
```diff
@@ -0,0 +1,33 @@
+rootProject.name = 'Keybase'
+
+apply from: '../node_modules/react-native-unimodules/gradle.groovy'
+includeUnimodulesProjects([modulesPaths: ['../../node_modules']])
```
Comment (chrisnojima) [inline shared/android/settings.gradle]: still `../..`?

File: shared/android/settings.gradle [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+include ':react-native-camera'
+project(':react-native-camera').projectDir = new File(rootProject.projectDir,   '../node_modules/react-native-camera/android')
+include ':react-native-image-picker'
+project(':react-native-image-picker').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-image-picker/android')
+include ':react-native-fetch-blob'
+project(':react-native-fetch-blob').projectDir = new File(rootProject.projectDir, '../node_modules/rn-fetch-blob/android')
+include ':react-native-contacts'
+project(':react-native-contacts').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-contacts/android')
+// include ':react-native-fast-image'
+// project(':react-native-fast-image').projectDir = new File(rootProject.projectDir, '../../node_modules/react-native-fast-image/android')
```
Comment (chrisnojima) [inline shared/android/settings.gradle]: lets update the path in this comment in case we uncomment and its not working and its confusing

**Comment by chrisnojima** [comment]
- [ ] merge in recent jsc changes (https://github.com/keybase/client/pull/17806) since the files moved

File: shared/android/settings.gradle (hunk 1 above)
Comment (buoyad) [reply shared/android/settings.gradle]: yeah, we actually don't need it because that's what it defaults to

**Comment by buoyad** [comment]
@chrisnojima mind if we thumb this and merge down before merging jsc changes? storybook is good on mobile :+1:

**Comment by chrisnojima** [comment]
Jsc is already merged

**Pull request merged**